odoo/models.py

```diff
@@ -204,6 +204,9 @@ def __repr__(self):
             "<NewId 0x%x>" % id(self)
         )
 
+    def __str__(self):
+        return "new_id_0x%x" % id(self)
```

Comment:
@rco-odoo 
Do you validate this. We need it to avoid having spaces in some css classes when doing `<t t-esc=company.id />` when company is a virtual record

Reply:
@Feyensv

Reply:
@kebeclibre how did you manage to inject new ids in templates?  That does not make sense.
Real programmers use real records :laughing:

Reply:
> how did you manage to inject new ids in templates? That does not make sense.
> Real programmers use real records

If I'm not wrong, kanban can contain new records (in x2m fields) and support qweb.  It should probably happen in this case :thinking:

Reply:
@rco-odoo 
:D hahahaha
anywhooooo

The use case is:
- a wizard allows to modify fields on the company (there is not any real records in this case)
- some onchange allows to re-render the template of a report according to the wizard's values (there is no real records either.....)
- the values in db are irrelevant anyway (it is a transient model and the values of the company are not the ones we are manipulating)

Reply:
@kebeclibre okayyy, now it makes sense to me.
Have you tried to use `company._origin.id` instead of `company.id`?
`company._origin` is the real record corresponding to `company`, and is empty if there is no such record.
With that trick, you will always get a real id or `False`.

Reply:
@rco-odoo The new ids are here used *inside* css class name and html class names. And as explained above, we never have read ids since it's not yet store in database. But having False as result would not cut it.